diavlos/src/service/service.py

```diff
@@ -426,7 +447,12 @@ def add(self, name, fields):
                 page.edit(te.wikitext())
                 page_name = page.page_title
                 page_full_name = page.name
-                result = self._service_dict(page_name, page_full_name, te)
+                current_revision = page.revisions(limit=1, dir='older').next()
+                latest_update_date = datetime.utcfromtimestamp(
+                    mktime(current_revision['timestamp'])).isoformat()
+                page_id = page._info['pageid']
+                result = self._service_dict(
+                    page_name, page_full_name, latest_update_date, page_id, te)
```

Comment:
Right now _service_dict requires the following parameters `page_name, page_full_name, latest_update_date, page_id, te`. However, creating these so as to pass them as arguments leads to unnecessary repeating (e.g. line 406 and 450 are the same). Better refactor `_service_dict` once more, so that it accepts only a `page` object: `_service_dict(page)` and everything is created inside the method.

Reply:
Sure, I will do that. Iv close this PR to squash commits and create a new one with refactoring.